src/main/scala/rocketchip/Periphery.scala

```diff
@@ -17,6 +17,11 @@ import rocket.XLen
 import scala.math.max
 import coreplex._
 
+case object MIFAddrBits extends Field[Int]
+case object MIFDataBits extends Field[Int]
+case object MIFTagBits extends Field[Int]
+case object MIFDataBeats extends Field[Int]
+
```

Comment:
As usual there's a chicken and egg question about whether the outside drives the inside or the inside drives the outside. I agree with splitting MIFDataBits into two parameters (and eliminate the other ones?). However, I think ExtMMIOWidth should just be forced to always be p(PeripheryBusKey).beatBytes...